components/segmenter/src/complex/lstm/mod.rs

```diff
@@ -155,24 +155,19 @@ impl<'l> LstmSegmenter<'l> {
                         return self.dic.len() as u16;
                     };
 
-                    // The maximum UTF-8 size of a grapheme cluster seems to be 41 bytes
-                    let mut i = 0;
-                    let mut buf = [0; 41];
-
-                    #[allow(clippy::unwrap_used)]
-                    // debug_asserting whether my assumption is correct
-                    decode_utf16(grapheme_cluster.iter().copied()).for_each(|c| {
-                        debug_assert!(i < 37);
-                        i += c
-                            .unwrap_or(REPLACEMENT_CHARACTER)
-                            .encode_utf8(&mut buf[i..])
-                            .len()
-                    });
-
-                    #[allow(clippy::unwrap_used)]
-                    // debug_asserting whether my assumption is correct
                     self.dic
-                        .get_copied(UnvalidatedStr::from_bytes(&buf[..i]))
+                        .get_copied_by(|key| {
+                            key.as_bytes().iter().copied().cmp(
```

Comment:
Suggestion (optional): UTF-8 byte order is equivalent to UTF-32 order; it may be cleaner/smaller/faster code if you compared an iterator of `char` rather than an iterator of `u8`.

Separately, I'm starting to get a bit worried that the grapheme cluster code may bloat the code size of the LSTM segmenter. We should probably delete it at some point. Maybe as part of the next round of ML model upgrades.

Reply:
The keys are `UnvalidatedStr`s aka `[u8]`, we cannot use them as anything that's strongly typed.

Reply:
There is https://docs.rs/utf8_iter/latest/utf8_iter/ for that

Reply:
We don't like deps

Reply:
Segmenter already depends on `utf8_iter`.
https://github.com/unicode-org/icu4x/blob/b6c4018a736e79790898c5b91ff3ab25d33192c2/components/segmenter/Cargo.toml#L35